src/redis_disk.cc

```diff
@@ -0,0 +1,146 @@
+                                          Slice subkeyright) {
+    std::string prefix_key, next_version_prefix_key;
+    InternalKey(ns_key, subkeyleft, metadata.version,
+                storage_->IsSlotIdEncoded()).Encode(&prefix_key);
+    InternalKey(ns_key, subkeyright, metadata.version + 1,
+                storage_->IsSlotIdEncoded()).Encode(&next_version_prefix_key);
+    auto key_range = rocksdb::Range(prefix_key, next_version_prefix_key);
+    uint64_t tmp_size = 0;
+    rocksdb::Status s = db_->GetApproximateSizes(this->option, column_family,
+                                                 &key_range, 1, &tmp_size);
+    if (!s.ok()) return s;
+    *key_size += tmp_size;
+    return rocksdb::Status::OK();
+}
+rocksdb::Status Disk::GetStringSize(const Slice &user_key, uint64_t *key_size) {
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    auto key_range = rocksdb::Range(Slice(ns_key), Slice(ns_key + static_cast<char>(0)));
+    return db_->GetApproximateSizes(this->option, metadata_cf_handle_, &key_range, 1, key_size);
+}
+
+rocksdb::Status Disk::GetHashSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    HashMetadata metadata(false);
+    rocksdb::Status s = Database::GetMetadata(kRedisHash, ns_key, &metadata);
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    return this->GetApproximateSizes(metadata, ns_key,
+                                     storage_->GetCFHandle(Engine::kSubkeyColumnFamilyName),
+                                     key_size);
+}
+
+
+rocksdb::Status Disk::GetSetSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    HashMetadata metadata(false);
+    rocksdb::Status s = Database::GetMetadata(kRedisSet, ns_key, &metadata);
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    return this->GetApproximateSizes(metadata, ns_key,
+                                     storage_->GetCFHandle(Engine::kSubkeyColumnFamilyName),
+                                     key_size);
+}
+
+rocksdb::Status Disk::GetListSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    ListMetadata metadata(false);
+    rocksdb::Status s = Database::GetMetadata(kRedisList, ns_key, &metadata);
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    std::string buf;
+    PutFixed64(&buf, metadata.head);
+    return this->GetApproximateSizes(metadata, ns_key,
+                                     storage_->GetCFHandle(Engine::kSubkeyColumnFamilyName),
+                                     key_size, buf);
+}
+
+rocksdb::Status Disk::GetZsetSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
```

Comment:
To be consistent with the rest of Kvrock's code, please use the indentation of 2 spaces instead of 4.

Reply:
> To be consistent with the rest of Kvrock's code, please use the indentation of 2 spaces instead of 4.

Thank you. It has been modified.